servant-foreign/src/Servant/Foreign/Internal.hs

```diff
@@ -120,6 +122,9 @@ type family Elem (a :: *) (ls::[*]) :: Constraint where
   Elem a (a ': list) = ()
   Elem a (b ': list) = Elem a list
 
+class HasForeignType a where
+    typeFor :: Proxy a -> ForeignType
```

Comment:
Funny thing is I've actually thought about that, but I've somehow assumed that instances specific to a backend would be hidden internally. But yeah, being explicit is better.